nav2_dwb_controller/dwb_critics/src/rotate_to_goal.cpp

```diff
@@ -76,16 +85,21 @@ double RotateToGoalCritic::scoreTrajectory(const dwb_msgs::msg::Trajectory2D & t
   }
 
   // If we're sufficiently close to the goal, any transforming velocity is invalid
-  if (fabs(traj.velocity.x) > EPSILON || fabs(traj.velocity.y) > EPSILON) {
+  if (fabs(traj.velocity.x) > EPSILON) {
     throw nav_core2::IllegalTrajectoryException(name_, "Nonrotation command near goal.");
   }
+  if (!ignore_y_velocity_) {
+    if (fabs(traj.velocity.y) > EPSILON) {
+      throw nav_core2::IllegalTrajectoryException(name_, "Nonrotation command near goal.");
+    }
+  }
 
   if (traj.poses.empty()) {
     throw nav_core2::IllegalTrajectoryException(name_, "Empty trajectory.");
   }
 
   double end_yaw = traj.poses.back().theta;
-  return angles::shortest_angular_distance(end_yaw, goal_yaw_);
+  return fabs(angles::shortest_angular_distance(end_yaw, goal_yaw_));
```

Comment:
Good catch, had to pull up the `angles.h` file myself to make sure that wasn't always positive.

Reply:
It wasn't so much a good catch as much as everything blew up without this change. :-)